src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1900,105 +1900,162 @@ let rec TryTranslateComputationExpression
             bindDebugPoint = spBind
             isUse = true
             isFromSource = isFromSource
-            pat = SynPat.Named(ident = SynIdent(id, _); isThisVal = false) as pat
+            pat = pat
             rhs = rhsExpr
-            andBangs = []
-            body = innerComp
-            trivia = { LetOrUseBangKeyword = mBind })
-        | SynExpr.LetOrUseBang(
-            bindDebugPoint = spBind
-            isUse = true
-            isFromSource = isFromSource
-            pat = SynPat.LongIdent(longDotId = SynLongIdent(id = [ id ])) as pat
-            rhs = rhsExpr
-            andBangs = []
+            andBangs = andBangs
             body = innerComp
             trivia = { LetOrUseBangKeyword = mBind }) ->
+            match pat with
+            | SynPat.Named(ident = SynIdent(id, _); isThisVal = false)
+            | SynPat.LongIdent(longDotId = SynLongIdent(id = [ id ])) when andBangs.IsEmpty ->
+                if ceenv.isQuery then
+                    error (Error(FSComp.SR.tcBindMayNotBeUsedInQueries (), mBind))
 
-            if ceenv.isQuery then
-                error (Error(FSComp.SR.tcBindMayNotBeUsedInQueries (), mBind))
+                if
+                    isNil (
+                        TryFindIntrinsicOrExtensionMethInfo
+                            ResultCollectionSettings.AtMostOneResult
+                            cenv
+                            ceenv.env
+                            mBind
+                            ceenv.ad
+                            "Using"
+                            ceenv.builderTy
+                    )
+                then
+                    error (Error(FSComp.SR.tcRequireBuilderMethod ("Using"), mBind))
```

Comment:
Nitpicking: the parens around `"Using"` don't seem to be required.

Reply:
Thanks.